setuptools/dist.py

```diff
@@ -67,7 +68,7 @@ def get_metadata_version(self):
     return mv
 
 
-def read_pkg_file(self, file):
+def read_pkg_file(self, file):  # noqa: C901
```

Comment:
These complexity checks are here for a reason. Can you devise a solution that avoids the complexity?

Reply:
Yeah, it would be possible to refactor this function. I'm just not sure it's the right thing to do here. Currently at least, we got all parsing code in one place. Refactor would probably spilt this into multiple functions.

Reply:
And this is the intended outcome of us having the complexity check enabled: adding noqa effectively ruins the expectation that we keep the code simple at all times. If you don't like doing the refactoring (rightfully) within this PR, please consider submitting a separate one and then rebasing on top of that.

Reply:
I can do the refactoring here, that's not the point. What I was trying to say is that I don't believe this to be a good idea. IMO leaving it as it is at the moment is easier to understand than any refactored version would be. That's why I added the `noqa` for only this specific function.

If you still want me to refactor it, please let me know and I'll do it.

Reply:
It's already quite complex. Adding `noqa` would enable people to make it worse over time without alerting them about the problem. So I think that having refactoring can be a good idea if performed properly.

If you look closer, all of these internal helper functions could be pure functions with just one extra argument allowing them to be kept outside of this function body.
OTOH, the problem here is that the abstractions this function operates are not of the same level. Why the data accessors do translation/normalization and also mapping to this object attributes?

It could be easier to move the call to `message_from_file()` into a separate function and have all the data normalization there. Meaning that here you'd just call that helper and get all the data ready to use.